packages/@aws-cdk-testing/framework-integ/test/aws-ecr/test/integ.lifecycle-rule-class.ts

```diff
@@ -0,0 +1,93 @@
+import * as ecr from 'aws-cdk-lib/aws-ecr';
+import * as cdk from 'aws-cdk-lib';
+import { IntegTest } from '@aws-cdk/integ-tests-alpha';
+
+/**
+ * Integration test for LifecycleRuleClass with L1 construct usage
+ *
+ * This test validates:
+ * 1. LifecycleRuleClass can be instantiated with various configurations
+ * 2. toJSON() method produces correct CloudFormation-compatible JSON
+ * 3. Integration with CfnRepositoryCreationTemplate works correctly
+ * 4. CloudFormation template generation is accurate
+ */
+
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'LifecycleRuleClassTest');
+
+// Test 1: Basic LifecycleRuleClass with CfnRepositoryCreationTemplate
+const basicLifecycleRule = new ecr.LifecycleRuleClass({
+  rulePriority: 1,
+  maxImageCount: 5,
+  tagStatus: ecr.TagStatus.TAGGED,
+  tagPrefixList: ['prod'],
+});
+
+new ecr.CfnRepositoryCreationTemplate(stack, 'BasicTemplate', {
+  appliedFor: ['PULL_THROUGH_CACHE'],
+  prefix: 'basic-test',
+  lifecyclePolicy: JSON.stringify({
+    rules: [basicLifecycleRule.toJSON()],
```

Comment:
If you're asking a user to do the conversion explicitly, then it might as well have been a function.